workflow-testing/src/main/java/com/squareup/workflow1/testing/WorkerTester.kt

```diff
@@ -54,67 +62,78 @@ public interface WorkerTester<T> {
 /**
  * Test a [Worker] by defining assertions on its output within [block].
  */
+@OptIn(ExperimentalCoroutinesApi::class)
 public fun <T> Worker<T>.test(
   timeoutMs: Long = DEFAULT_TIMEOUT_MS,
   block: suspend WorkerTester<T>.() -> Unit
 ) {
-  runBlocking {
-    supervisorScope {
-      val channel: ReceiveChannel<T> = run().produceIn(this + Unconfined)
-
+  runTest {
+    // Use a Turbine which consumes outputs/errors from an underlying channel.
+    run().test(
+      timeout = timeoutMs.toDuration(MILLISECONDS)
+    ) {
       val tester = object : WorkerTester<T> {
-        override suspend fun nextOutput(): T = channel.receive()
+        override val testCoroutineScheduler: TestCoroutineScheduler = testScheduler
+
+        override suspend fun nextOutput(): T = awaitItem()
 
         override fun assertNoOutput() {
-          // isEmpty returns false if the channel is closed.
-          if (!channel.isEmpty && !channel.isClosedForReceive) {
+          try {
+            expectNoEvents()
+          } catch (e: AssertionError) {
             throw AssertionError("Expected no output to have been emitted.")
           }
         }
 
         override suspend fun assertFinished() {
```

Comment:
This mapping is the most complicated in order to get the same error messages in the AssertionError as before.

Reply:
Seems worth saying so in a comment, e.g. "Complicated because these messages predate our Turbine integration and we wanted to keep them stable."